src/mpi/comm/comm_split_type.c

```diff
@@ -154,6 +154,8 @@ static int node_split_processor(MPIR_Comm * comm_ptr, int key, hwloc_obj_type_t
 
     obj_containing_cpuset =
         hwloc_get_obj_covering_cpuset(MPIR_Process.topology, MPIR_Process.bindset);
+    if (obj_containing_cpuset == NULL)
```

Comment:
In that case will this be NULL?  Shouldn't there be at least one object that covers any cpuset?

Reply:
Unlikely that it ever will be since the call to node_split_processor  is guarded by the return value of the cpu bindset query's return value in the first place. This is probably an overly cautious piece of code.

Reply:
Your comment doesn't answer my question.

Reply:
sorry, i updated my response.

Reply:
I don't think this PR is correct, then.  If it should never be NULL, then you can add an assert to make sure it is not.  But going to `fn_exit` means that it's valid behavior.

Reply:
Ok, I will replace the NULL check with an assert. Thank you.